### ISSUE TITLE
The example `with-webassembly` fails with invalid hook

### ISSUE THREAD

**Issue by ignacio-gc** [vercel/next.js#18867]
# Bug report

## Describe the bug

The example `with-webassembly` fails with with: `Error: Invalid hook call.`

## To Reproduce

`npx create-next-app --example with-webassembly with-webassembly-app`

install Rust

`npm run build-rust`

`npm run dev`

## Expected behavior

`npm run dev` starts without error

## System information

- OS: Archlinux
- Chrome
 Version 86.0.4240.183 (Official Build) Arch Linux (64-bit)
- Version of Next.js: 10.0.1
- Version of Node.js: 15.0.1

**Comment by davidsonsns** [comment]
Found only the example `with-webassembly`, without `-app`. Does it right?

If so, tried to reproduce it but I didn't get any error. Noticed that the only different point is the OS, I'm using a macOS Catalina.

**Comment by ignacio-gc** [comment]
Yes, with no "app", sorry. Do you compiled the Rust code?

**Comment by davidsonsns** [comment]
If you mean about run `npm run build-rust`(very newbie about rust here), initially I didn't run it. But, to try reproducing your bug reported I installed `rust` and `wasm32-unknown-unknown` target. After that, I ran the command `build-rust` and `npm run dev`.
After all, it still works well here. So, could you accordingly add your steps to reproduce it in detail? More specifically on the section `To Reproduce`, please. With that, another guy with more skills about `rust` can help you better.

**Comment by ignacio-gc** [comment]
I have installed Rust nightly and wasm32-unknow-unknow target and I followed the steps of the example.

**Comment by davidsonsns** [comment]
Thanks for adding the steps in detail. It was very helpful because I was running on my forked `next.js` project directly and another point is that I noticed that was running it using `yarn`.

Once you wrote your commands with `npm run ...`, I tried to do it and got the same issue reported.
Looking at the error log I saw the link https://reactjs.org/link/invalid-hook-call. And running `npm ls react` I saw two versions of `react`:

```sh
with-webassembly@ /Users/.../with-webassembly-app
├─┬ next@10.0.1
│ ├─┬ @next/react-dev-overlay@10.0.1
│ │ └── react@16.14.0 deduped
│ ├── react@17.0.1
│ ├─┬ styled-jsx@3.3.1
│ │ └── react@16.14.0 deduped
│ └─┬ use-subscription@1.5.0
│   └── react@17.0.1 deduped
├─┬ react-dom@16.14.0
│ └── react@16.14.0 deduped
└── react@16.14.0
```

In short, I got it fixed using `yarn` or replacing the packages `react` and `react-dom` to version `^17.0.0` both.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
